### ISSUE TITLE
20 warnings building repo

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefxlab#2585]
```
D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj : warning NU1701: Package 'codetitans-json 1.8.3' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.1'. This package may not be fully compatible with your project.
D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj : warning NU1701: Package 'jayrock-json 0.9.16530.1' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.1'. This package may not be fully compatible with your project.
D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj : warning NU1701: Package 'System.Text.Json 2.0.0.11' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.1'. This package may not be fully compatible with your project.
Utf8BufferEncoder.cs(32,23): warning CS0618: 'UrlEncoder.Decode(ReadOnlySpan<byte>, Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.Decode' [D:\j\workspace\windows_nt_re---4abe806d\src\System.Text.Encodings.Web.Utf8\System.Text.Encodings.Web.Utf8.csproj]
Utf8BufferEncoder.cs(39,23): warning CS0618: 'UrlEncoder.DecodeInPlace(Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.DecodeInPlace' [D:\j\workspace\windows_nt_re---4abe806d\src\System.Text.Encodings.Web.Utf8\System.Text.Encodings.Web.Utf8.csproj]
Utf8BufferEncoder.cs(14,17): warning CS0618: 'UrlEncoder.TryEncode(ReadOnlySpan<byte>, Span<byte>, out int)' is obsolete: 'Use UrlEncoder.Utf8.Encode' [D:\j\workspace\windows_nt_re---4abe806d\src\System.Text.Encodings.Web.Utf8\System.Text.Encodings.Web.Utf8.csproj]
System\Buffers\Native\NativeMemoryPool.cs(27,40): warning CS0108: 'NativeMemoryPool.Shared' hides inherited member 'MemoryPool<byte>.Shared'. Use the new keyword if hiding was intended. [D:\j\workspace\windows_nt_re---4abe806d\src\System.Buffers.Experimental\System.Buffers.Experimental.csproj]
UnescapeUrlPathInPlaceTests.cs(19,23): warning CS0618: 'UrlEncoder.DecodeInPlace(Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.DecodeInPlace' [D:\j\workspace\windows_nt_re---4abe806d\tests\System.Text.Encodings.Web.Utf8.Tests\System.Text.Encodings.Web.Utf8.Tests.csproj]
UnescapeUrlPathTests.cs(19,23): warning CS0618: 'UrlEncoder.Decode(ReadOnlySpan<byte>, Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.Decode' [D:\j\workspace\windows_nt_re---4abe806d\tests\System.Text.Encodings.Web.Utf8.Tests\System.Text.Encodings.Web.Utf8.Tests.csproj]
UnescapeUrlPathTests.cs(37,17): warning CS0618: 'UrlEncoder.Decode(ReadOnlySpan<byte>, Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.Decode' [D:\j\workspace\windows_nt_re---4abe806d\tests\System.Text.Encodings.Web.Utf8.Tests\System.Text.Encodings.Web.Utf8.Tests.csproj]
System.Buffers\Reader_Search.cs(23,26): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_Advance.cs(39,31): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_ParseInt.cs(21,13): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_ParseInt.cs(24,26): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_Construction.cs(25,26): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
CSC : warning CS8002: Referenced assembly 'Jayrock.Json, Version=0.9.16530.0, Culture=neutral, PublicKeyToken=null' does not have a strong name. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
CSC : warning CS8002: Referenced assembly 'LitJSON, Version=0.13.0.0, Culture=neutral, PublicKeyToken=null' does not have a strong name. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
CSC : warning CS8002: Referenced assembly 'SpanJson, Version=1.0.7.0, Culture=neutral, PublicKeyToken=null' does not have a strong name. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
CSC : warning CS8002: Referenced assembly 'Jil, Version=2.16.0.0, Culture=neutral, PublicKeyToken=null' does not have a strong name. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
CSC : warning CS8002: Referenced assembly 'System.Text.Json, Version=2.0.0.11, Culture=neutral, PublicKeyToken=null' does not have a strong name. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
```

It looks like at least some can be fixed. @ahsonkhan

**Comment by ahsonkhan** [comment]
```text
System.Buffers\Reader_Search.cs(23,26): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_Advance.cs(39,31): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_ParseInt.cs(21,13): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_ParseInt.cs(24,26): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
System.Buffers\Reader_Construction.cs(25,26): warning CS0436: The type 'BufferUtilities' in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs' conflicts with the imported type 'BufferUtilities' in 'System.Buffers.ReaderWriter.Tests, Version=0.1.0.0, Culture=neutral, PublicKeyToken=c6389b02d38f98e6'. Using the type defined in 'D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\..\System.Buffers.Primitives.Tests\BufferUtilities.cs'. [D:\j\workspace\windows_nt_re---4abe806d\tests\Benchmarks\Benchmarks.csproj]
```

cc @JeremyKuhne 

Regarding:
- warning NU1701
- warning CS8002

I am not sure if we should be suppressing these. Should we? We are knowingly including references to these external JSON libaries for benchmarking.

These can be fixed, yes. We had marked them obsolete temporarily a while back to help upstream consumers change the API usage. We can remove the obsolete APIs now.

```text
Utf8BufferEncoder.cs(32,23): warning CS0618: 'UrlEncoder.Decode(ReadOnlySpan<byte>, Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.Decode' [D:\j\workspace\windows_nt_re---4abe806d\src\System.Text.Encodings.Web.Utf8\System.Text.Encodings.Web.Utf8.csproj]
Utf8BufferEncoder.cs(39,23): warning CS0618: 'UrlEncoder.DecodeInPlace(Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.DecodeInPlace' [D:\j\workspace\windows_nt_re---4abe806d\src\System.Text.Encodings.Web.Utf8\System.Text.Encodings.Web.Utf8.csproj]
Utf8BufferEncoder.cs(14,17): warning CS0618: 'UrlEncoder.TryEncode(ReadOnlySpan<byte>, Span<byte>, out int)' is obsolete: 'Use UrlEncoder.Utf8.Encode' [D:\j\workspace\windows_nt_re---4abe806d\src\System.Text.Encodings.Web.Utf8\System.Text.Encodings.Web.Utf8.csproj]
UnescapeUrlPathInPlaceTests.cs(19,23): warning CS0618: 'UrlEncoder.DecodeInPlace(Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.DecodeInPlace' [D:\j\workspace\windows_nt_re---4abe806d\tests\System.Text.Encodings.Web.Utf8.Tests\System.Text.Encodings.Web.Utf8.Tests.csproj]
UnescapeUrlPathTests.cs(19,23): warning CS0618: 'UrlEncoder.Decode(ReadOnlySpan<byte>, Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.Decode' [D:\j\workspace\windows_nt_re---4abe806d\tests\System.Text.Encodings.Web.Utf8.Tests\System.Text.Encodings.Web.Utf8.Tests.csproj]
UnescapeUrlPathTests.cs(37,17): warning CS0618: 'UrlEncoder.Decode(ReadOnlySpan<byte>, Span<byte>)' is obsolete: 'Use UrlDecoder.Utf8.Decode' [D:\j\workspace\windows_nt_re---4abe806d\tests\System.Text.Encodings.Web.Utf8.Tests\System.Text.Encodings.Web.Utf8.Tests.csproj]
```

I will have to figure out how we can circumvent this warning. Not sure atm.
```text
System\Buffers\Native\NativeMemoryPool.cs(27,40): warning CS0108: 'NativeMemoryPool.Shared' hides inherited member 'MemoryPool<byte>.Shared'. Use the new keyword if hiding was intended. [D:\j\workspace\windows_nt_re---4abe806d\src\System.Buffers.Experimental\System.Buffers.Experimental.csproj]
```

**Comment by danmosemsft** [comment]
We should suppress, with open issues of necessary, any warning we do not plan to promptly fix. Similar for introducing new warnings. We should  always be totally clean or new ones are hidden. (I know corefx isn't currently)

**Comment by danmosemsft** [comment]
@ahsonkhan I have addressed all of these in #2600 except the 6 UrlEncoder ones. It is not clear to me what fix you expect for those - some non obsolete methods are implemented in terms of the obsolete ones. Can you please fix those so we can get clean? Then we can enable warning-as-error perhaps.

**Comment by ahsonkhan** [comment]
Fixed in https://github.com/dotnet/corefxlab/pull/2600 / https://github.com/dotnet/corefxlab/pull/2602. Closing.
